### ISSUE TITLE
RFE: OpenAPI spec

### ISSUE THREAD

**Issue by ananace** [DataFire/OneDB#4]
Could an OpenAPI spec be written / provided for the REST APIs used in the communication with / between OneDB servers?

Such a thing would help immensely in developing applications using OneDB in other languages than JavaScript / ECMAScript and its derivatives. Native desktop and mobile applications, etc.

Additionally, having an available server communication spec could help with developing OneDB servers in other languages as well, which could also help with - in the future - building hostable appliances and the like utilizing OneDB.

**Comment by bobby-brennan** [comment]
Yes, this is definitely doable, and something we should aim for near-term.

There will likely be two options here:
* A generic spec, which will have paths like `GET /data/{namespace}/{type}/{id}`, with generic JSON schemas for requests/responses
* The ability to generate a spec for a specific set of namespaces, like the `chat` namespace, which will have paths like `GET /data/chat/message/{id}`, and the JSON schemas specific to the `chat` namespace
